tbot/machine/machine.py

```diff
@@ -194,3 +206,31 @@ def _init_machine(self) -> typing.ContextManager:
             More docs for this ...
         """
         raise tbot.error.AbstractMethodError()
+
+
+class PreConnectInitializer(Machine):
+    """
+    Base-class for pre-connection initializer context. This initializer is run
+    before connecting to the machine, before Initializer._init_machine().
+    """
```

Comment:
Python doc strings (at least the way they are used in tbot) should start with a single line short description, then a blank line and then an (optional) longer description.

Also, to link to a different method, use the following syntax:

```rst
:py:meth:`tbot.machine.Initializer._init_machine`
```